v-next/hardhat/src/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -23,15 +23,15 @@ const httpNetworkUserConfigSchema = z.object({
   from: z.optional(z.string()),
   gas: z.optional(
     unionType(
-      [z.literal("auto"), z.number().int(), z.bigint()],
-      "Expected 'auto', number, or bigint",
+      [z.literal("auto"), z.number().int().safe(), z.bigint()],
```

Comment:
```suggestion
      [z.literal("auto"), z.number().int().positive().safe(), z.bigint()],
```

I think we are missing the `.positive()`.

We can also create a `gasSchema` to avoid duplicating it.

Finally, I think this would be better with a `conditionalUnionType`, as very few people understand what a safe number is. A `conditionalUnionType` will give more specific errors:

```
> z.number().int().positive().safe().safeParse(-1).error.errors[0].message
'Number must be greater than 0'
> z.number().int().positive().safe().safeParse(1.2).error.errors[0].message
'Expected integer, received float'
> z.number().int().positive().safe().safeParse(1e18).error.errors[0].message
'Number must be less than or equal to 9007199254740991'
```

Reply:
I've updated the code to use a `gasSchema`, and replaced the `unionType` with a `conditionalUnionType`. I've also added the `.positive` check for both `int` and `bigint`. See [`87f5940` (#5762)](https://github.com/NomicFoundation/hardhat/pull/5762/commits/87f5940a467955eab4663db12716030c95c895fe)